### ISSUE TITLE
Unexpected Babel configuration behavior

### ISSUE THREAD

**Issue by d3dc** [sharegate/craco#36]
[The babel config override](https://github.com/sharegate/craco/blob/master/packages/craco/lib/features/babel.js#L76) will override the loaderOptions, presets, and plugins for [both of the `babel-loader` rules CRA has.](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/webpack.config.js#L396)

It seems like the second loader shouldn't receive the presets and plugins by default, since the second loader is meant for:
> // Process any JS outside of the app with Babel.
   // Unlike the application JS, we only compile the standard ES features.

I would argue the behavior is unexpected since only files in `src/` are supposed to get `babel-preset-react-app` and previously tools such as [`react-app-rewired` only modified the first loader](https://github.com/timarney/react-app-rewired/blob/master/packages/react-app-rewired/index.js#L17)

**Comment by patricklafrance** [comment]
Hi @d3dc 

Thank you for your feedback.

Interesting, I haven't notice that. 

May I ask what  side effects you are experimenting since craco updates both babel-loaders?

Do you see any use cases were someone would require to apply presets or plugins to both babel loaders?

Thank you

**Comment by d3dc** [comment]
The second loader is intended to polyfill code for browsers with `react-app-polyfills`. I don't think there _should_ be too many side-effects.

The biggest is performance. Each plugin will traverse the AST for each import - and this happens for all of `node_modules` with the second loader.

IMO, the second loader is the "polyfill loader". I could see a use case where someone would want to apply a babel preset to all code to support an "environment".

&nbsp;

My use case is a little out of the ordinary - I'm trying to package my whole toolkit with a customized craco config and uncompiled library code. When using lerna's `bootstrap`, my babel config applies differently than when using the packaged code, because the symlink resolves. My fix is just to check which loader is being modified in my `loaderOptions` function.

**Comment by d3dc** [comment]
So to clarify, anything outside of source ordinarily only gets:
- `@babel/preset-env`
- `@babel/plugin-transform-runtime`
- `@babel/plugin-syntax-dynamic-import`

This does seem odd to automatically add all new plugins to.

It is already pretty convenient to use  a `loaderOptions` function:
```
  loaderOptions: opts => {
    if (!opts.customize) {
      // external files
      return opts
    }

    return {
      ...opts,
      ...customization
    }
```

**Comment by patricklafrance** [comment]
Hi @d3dc 

This is indeed an easy fix if you want to customize what's being configured for the files outside of the `src/` directory.

I might wait before updating any code for this one since I don't want to break anyone setup.

How could craco's configuration be modified to accommodate your usecase?

Maybe an `applyToExternalFiles`  boolean option with a default value of `false` ? Not sure though if it wouldn't be more confusing than helping?

```javascript
{
   babel: {
        presets: [],
        plugins: [],
        loaderOptions: { }
        applyToExternalFiles: false
   }
}
```

**Comment by d3dc** [comment]
I ended up writing a custom function for my babel configuration. Its pretty close to the original, with two things:
- it only applies settings to the "customize" loader
- it modifies the same loader to include specific external libs

```js
  overrideWebpackConfig({ webpackConfig, cracoConfig }) {
   ...
    const {
      externalLibs,
      presets,
      plugins,
      loaderOptions,
    } = cracoConfig.customBabel

    matches.forEach(({ loader }) => {
      if (loader.options.customize) {
        Object.assign(loader.options, loaderOptions, {
          presets: [...loader.options.presets, ...presets],
          plugins: [...loader.options.plugins, ...plugins],
        })

        if (externalLibs) {
          loader.include = (Array.isArray(loader.include)
            ? loader.include
            : [loader.include]
          ).concat(externalLibs)

          loader.exclude = {
            test: loader.exclude || /node_modules/,
            not: externalLibs,
          }
        }
      }
    })

    return webpackConfig
  }
```

The boolean approach is probably better for react native libraries, as there could be frequent changes to the config with my way. I would think the performance would need testing before saying a whitelist is faster, but I also think a whitelist is faster.

We both used the word "external" but I wonder if "applyOutsideSrc" is simpler.

**Comment by patricklafrance** [comment]
Thanks for sharing.

I agree with you that `applyOutsideSrc` is more accurare.

**Comment by patricklafrance** [comment]
Will be fix in #49
